crates/core/src/test.rs

```diff
@@ -8073,6 +8073,29 @@ fn test_basic_md() {
     assert_yaml_snapshot!(results);
 }
 
+#[test]
+fn test_basic_php() {
+    let pattern = r#"
+    |language php
+    |
+    |`echo $str` => `echo $str;`
+    |"#
+    .trim_margin()
+    .unwrap();
+
+    let source = r#"echo "hello world""#.trim_margin().unwrap();
+    let file = "foo.php";
+
+    let context = ExecutionContext::default();
+    let language: TargetLanguage = PatternLanguage::Php.try_into().unwrap();
+    // println!("language: {language} pattern: {pattern}");
+
+    let pattern = src_to_problem(pattern, language).unwrap();
+    let results = pattern.execute_file(&RichFile::new(file.to_owned(), source), &context);
+    // println!("results: {results}");
+    assert_yaml_snapshot!(results);
+}
```

Comment:
don't use a snapshot test for this, instead assert that a rewrite occurs as expected using the `run_test_expected` function and `TestArgExpected` struct